test/generate.test.ts

```diff
@@ -149,6 +150,86 @@ describe('generate', function () {
     });
   });
 
+  describe('scaleType', () => {
+    it('should enumerate correct scaletype for quantitative field', () => {
+      const specQ = {
+        mark: Mark.POINT,
+        encodings: [
+          {
+            channel: Channel.X,
+            scale: {type: {values: [undefined, ScaleType.LOG]}},
+            field: 'Q',
+            type: Type.QUANTITATIVE
+          }
+        ]
+      };
+
+      const answerSet = generate(specQ, schema, stats);
+      assert.equal(answerSet.length, 2);
+      assert.equal((answerSet[0].getEncodingQueryByIndex(0).scale as ScaleQuery).type, undefined);
+      assert.equal((answerSet[1].getEncodingQueryByIndex(0).scale as ScaleQuery).type, ScaleType.LOG);
+    });
+
+    it('should enumerate correct scaletype for temporal field without timeunit', () => {
```

Comment:
Missing a case for temporal field with timeUnit (which should support `ordinal`)